src/formio/components/TextField.js

```diff
@@ -30,22 +32,41 @@ class TextField extends Formio.Components.components.textfield {
     return super.checkComponentValidity(data, dirty, row, {...options, async: true});
   }
 
-  setLocationData(postcode, house_number, key) {
-    get(`${this.options.baseUrl}location/get-street-name-and-city`, {postcode, house_number})
-      .then(result => {
-        if (result[key]) {
-          this.setValue(result[key]);
-        } else {
-          this.setValue('');
-        }
-      })
-      .catch(error => console.log(error));
+  /**
+   * Return a debounced method to look up and autocomplete the location data.
+   */
+  get setLocationData() {
+    if (!this._debouncedSetLocationData) {
+      this._debouncedSetLocationData = debounce((postcode, house_number, key) => {
+        get(`${this.options.baseUrl}location/get-street-name-and-city`, {postcode, house_number})
+          .then(result => {
+            if (result[key]) {
+              this.setValue(result[key]);
+            } else {
+              this.setValue('');
+            }
+          })
+          .catch(console.error);
+      }, LOCATION_AUTOCOMPLETE_DEBOUNCE);
+    } else {
+      this._debouncedSetLocationData.cancel();
+    }
+    return this._debouncedSetLocationData;
   }
 
   handleSettingLocationData(data) {
     const isValidHouseNumber = HOUSE_NUMBER_REGEX.test(data[this.component.deriveHouseNumber]);
     const isValidPostcode = POSTCODE_REGEX.test(data[this.component.derivePostcode]);
 
+    // Fill data if it is not set yet or if the field is readonly (i.e. Formio's disabled).
+    // Unrelated to the HTML 'disabled' attribute.
+    // See #1717 and #1832 - we only want to make API calls if there is useful work to
+    // be done.
+    const mayAutofillValue = !this.getValue() || this.component.disabled;
```

Comment:
Does this change the streetname/city if the user entered a house number/postcode and then changes it to another one?

Reply:
it doesn't - that's the biggest problem with this patch (and #1717). If they clear the field completely, then it will be autofilled again.

I'm thinking that we could probably streamline this in the future (read: another PR) where we keep track if the value was set automatically by autofill, or if it was set by the end-user (manually). In the first case, we could then update again even though there's a value set already, in the second case we leave it untouched.